automod/rules/spammentions.go

```diff
@@ -0,0 +1,38 @@
+package rules
+
+import (
+	appbsky "github.com/bluesky-social/indigo/api/bsky"
+	"github.com/bluesky-social/indigo/automod"
+)
+
+var _ automod.PostRuleFunc = SpamMentionsRule
```

Comment:
This wasn't in other rules that I looked at, but I'd like to suggest a "type assertion" like this as a convention.

I find they're good for self-documenting code in general.  They're a good hint at the top of a file about what's coming.  And the closer it is to the type's definition, the more immediately useful any compiler is at bringing you to the origin of the problem if the invariant fails.

Right now, the only other place this gets checked is in the `DefaultRules` constructor _if your rule is not commented out_.  So that's a tad... not fragile per se, but not as sturdy as it could be, either.

Reply:
Yeah! I'm a fan of those and have a couple in `indigo/atproto/crypto`. I would support having this be a norm in every rule.

For some reason I thought I had read a popular external style guide *against* this pattern, which was confusing, but maybe I was just confused. Uber likes it (for whatever Uber's opinion is worth):
https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance